cloudsync_onedrive.py

```diff
@@ -1387,15 +1425,17 @@ def _get_validated_namespace(self, ns_id: str):
                     raise CloudNamespaceError(f"Unknown site id: {ns_id}")
                 self._fetch_drives_for_site(site)
                 drive = self.__drive_by_id.get(ns_id)
-                if not drive:
-                    # check if it is a shared drive
+                if not drive and site == self._shared_with_me and not ids.is_shared and self._is_biz:
+                    # backwards compatibility for legacy shared folder namespaces (ODB only)
                     drive = next((d for d in self._shared_with_me.drives if d.drive_id == ids.drive_id), None)
+                    if drive:
+                        name = "/".join(drive.name.split("/")[:-1])
```

Comment:
`self.join()`

Reply:
this isn't really a path, its the drive name, formatted as <site-name>/<drive-name>/<path-to-root-folder>

since it is for display purposes only, a simple string join should be good enough (and I wanted to avoid any normalization that the path join func might do)

Reply:
Not related to the current review, the join function shouldn't do any normalization, other than slashes. I believe the join tests assert that.

Back on topic, I would ordinarily want any name that is for "display purposes only" to be called "display_name" rather than "name" in order to clarify and document that it is descriptive only, although I admit the scope of changing the name of the `name` property of the `Namespace` class is not the smallest change